### PR TITLE
Expose new metrics for memory usage in the cgroup.

### PR DESCRIPTION
**Pull request by DevinLeamy**
The metric "mem_kmem_usage_bytes" is the total kernel memory usage by
processes in the cgroup in bytes.

The metric "mem_kmem_tcp_usage_bytes" is the total memory usage for TCP
buffers in bytes.

### CODE REVIEW

File: include/mesos/mesos.proto [hunk 1]
```diff
@@ -1819,6 +1819,12 @@ message ResourceStatistics {
   // Total memory + swap usage. This is set if swap is enabled.
   optional uint64 mem_total_memsw_bytes = 37;
 
+  // Current kernel memory allocation
```
Comment (cf-natali) [inline include/mesos/mesos.proto:1822]: Nit: other comments end with a dot '.'.

File: include/mesos/v1/mesos.proto [hunk 2]
```diff
@@ -1783,6 +1783,12 @@ message ResourceStatistics {
   // Total memory + swap usage. This is set if swap is enabled.
   optional uint64 mem_total_memsw_bytes = 37;
 
+  // Current kernel memory allocation
+  optional uint64 mem_kmem_usage_bytes = 52;
```
Comment (cf-natali) [inline include/mesos/v1/mesos.proto:1787]: Is there a reason the valued don't match `include/mesos/mesos.proto`?

File: src/slave/containerizer/mesos/isolators/cgroups/subsystems/memory.cpp [hunk 3]
```diff
@@ -460,6 +460,26 @@ Future<ResourceStatistics> MemorySubsystemProcess::usage(
...
+    result.set_mem_kmem_usage_bytes(kmem_usage.get().bytes());
+  }
+
+  Result<Bytes> kmem_tcp_usage = cgroups::memory::kmem_tcp_usage_in_bytes(
+      hierarchy, cgroup);
+
+  if (kmem_tcp_usage.isError()) {
+    return Failure("Failed to parse 'memory.kmem.tcp.usage_in_bytes': " +
+                   kmem_tcp_usage.error());
+  } else if (kmem_usage.isSome()) {
```
Comment (cf-natali) [inline src/slave/containerizer/mesos/isolators/cgroups/subsystems/memory.cpp:479]: Should be `kmem_tcp_usage` here.

File: include/mesos/mesos.proto (hunk 1 above)
Comment (DevinLeamy) [reply include/mesos/mesos.proto]: Added periods.

File: src/slave/containerizer/mesos/isolators/cgroups/subsystems/memory.cpp (hunk 3 above)
Comment (DevinLeamy) [reply src/slave/containerizer/mesos/isolators/cgroups/subsystems/memory.cpp]: Good catch :)

File: include/mesos/v1/mesos.proto (hunk 2 above)
Comment (DevinLeamy) [reply include/mesos/v1/mesos.proto:1787]: There isn't a good reason. I've updated `mesos/mesos.proto` to match `mesos/v1/mesos.proto` (i.e. use the smaller field numbers).

**Comment by bmahler** [comment]
Mind squashing this locally on your end since you've got follow on commits here that should be part of the original single commit?

**Comment by DevinLeamy** [comment]
> Mind squashing this locally on your end since you've got follow on commits here that should be part of the original single commit?

Squashed.

**Pull request merged**